### PR TITLE
[MRG] Implementation the initial report agent

### PR DESCRIPTION
**Pull request by huangyz0918**
### **User description**
- [x] Fixed the integration re-integrating issue
- [x] build the first version of the agent prompt
- [x] integration with the user prompt and aggregate the knowledage

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Refactored `ReportAgent` to focus on generating progress reports, removing ML task suggestion functionalities.
- Enhanced integration logic for GitHub and Google Calendar, including checks for existing tokens.
- Updated model loading to use "gpt-4o" in the CLI.
- Improved Google Calendar integration with better error handling and date processing.
- Made minor formatting improvements for better code readability.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>reporter.py</strong><dd><code>Refactor ReportAgent for Progress Report Generation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

mle/agents/reporter.py

<li>Refactored <code>ReportAgent</code> to focus on generating progress reports.<br> <li> Removed unused imports and functions related to ML task suggestions.<br> <li> Added new methods for processing GitHub and Google Calendar data.<br> <li> Updated prompts and JSON output format for report generation.

</details>

  </td>
  <td><a href="https://github.com/MLSysOps/MLE-agent/pull/171/files#diff-940530770e98ef10492ff2db2bb6cb921d6ed0c52f9ceb63539685fb8cf319a4">+55/-112</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>cli.py</strong><dd><code>Enhance Integration Logic and Update Model Loading</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

mle/cli.py

<li>Updated GitHub and Google Calendar integration logic.<br> <li> Added checks for existing tokens before prompting for new ones.<br> <li> Changed model loading to use "gpt-4o".

</details>

  </td>
  <td><a href="https://github.com/MLSysOps/MLE-agent/pull/171/files#diff-346b4a2c637c537f5bd5dfbc35d6dcc4b7ec9fb95722b368aa2eebd55dc162bd">+24/-17</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>google_calendar.py</strong><dd><code>Refactor Google Calendar Integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

mle/integration/google_calendar.py

<li>Refactored Google Calendar login and event retrieval logic.<br> <li> Improved error handling and date processing.

</details>

  </td>
  <td><a href="https://github.com/MLSysOps/MLE-agent/pull/171/files#diff-9ed08b2ba29d54c99f63d94ca8e5013232a7f5909aae7d58c8a627523e109589">+32/-25</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Formatting</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>summarizer.py</strong><dd><code>Code Formatting Improvements</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

mle/agents/summarizer.py

- Minor formatting changes for better readability.

</details>

  </td>
  <td><a href="https://github.com/MLSysOps/MLE-agent/pull/171/files#diff-98ad8ed051584070f7d7d36729c0ae0792a281ae776d8eb4ffd4efcf9e4fe2c1">+5/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Miscellaneous</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>report.py</strong><dd><code>Remove Unused Import</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

mle/workflow/report.py

- Removed unused import `ask_text`.

</details>

  </td>
  <td><a href="https://github.com/MLSysOps/MLE-agent/pull/171/files#diff-647562765012a92e026a3b5281dd379532e4b9903e7657fc63dc2882633fe0e7">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: mle/agents/summarizer.py [hunk 1]
```diff
@@ -7,18 +7,20 @@
...
 class SummaryAgent:
 
-    def __init__(self, model, github_repo: str, github_token: str = None, console=None):
+    def __init__(self, model, github_repo: str, username: str, github_token: str = None, console=None):
         """
         SummaryAgent: summary the workspace provided by the user.
 
         Args:
             model: the model to use.
+            username: the Github username of the user.
```
Comment (HuaizhengZhang) [inline mle/agents/summarizer.py:16]: so this is just for GitHub summary?

File: mle/workflow/report.py [hunk 2]
```diff
@@ -26,8 +28,8 @@ def ask_data(data_str: str):
 
 def ask_github_token():
```
Comment (HuaizhengZhang) [inline mle/workflow/report.py:29]: should we put this util functions here?

very weried to me

**Review by HuaizhengZhang — APPROVED** [review]
small issues can be addressed this pr or next pr as you wish

File: mle/agents/summarizer.py (hunk 1 above)
Comment (HuaizhengZhang) [reply mle/agents/summarizer.py:16]: why not call this SummaryGitHub

then we can have `SummaryZoom`

Comment (huangyz0918) [reply mle/agents/summarizer.py:16]: so far, yes

File: mle/workflow/report.py (hunk 2 above)
Comment (huangyz0918) [reply mle/workflow/report.py:29]: Yeah, the whole Github auth and token-saving progress is a bit of a mess. We need to refactor it later

**Comment by leeeizhang** [comment]
Many thanks!

**Pull request merged**